otter/integration/tests/test_convergence.py

```diff
@@ -62,6 +63,130 @@ def _check_fds(_):
+        and does not overshoot in the process.
+
+        Further, we want to make sure the deleted servers are removed from the
+        CLB.
+
+        This variant assumes a scaling group's max_entities field exceeds that
+        for a CLB (as of this writing).  We use max of CLB + 25.
+        """
+        return self._perform_oobd_clb_test(50)
+    test_scaling_to_clb_max_after_oob_delete_type2.timeout = 1800
+
+    def _perform_oobd_clb_test(self, scaling_group_max_entities):
+        rcs = TestResources()
+
+        def create_clb_first():
+            self.clb = CloudLoadBalancer(pool=self.pool)
+            return (
+                self.identity.authenticate_user(rcs)
+                .addCallback(
+                    rcs.find_end_point,
+                    "otter", "autoscale", region,
+                    default_url='http://localhost:9000/v1.0/{0}'
+                ).addCallback(
+                    rcs.find_end_point,
+                    "nova", "cloudServersOpenStack", region
+                ).addCallback(
+                    rcs.find_end_point,
+                    "loadbalancers", "cloudLoadBalancers", region
+                ).addCallback(self.clb.start, self)
+                .addCallback(self.clb.wait_for_state, "ACTIVE", 600)
+            )
+
+        def then_test(_):
+            scaling_group_body = create_scaling_group_dict(
+                image_ref=image_ref, flavor_ref=flavor_ref,
+                use_lbs=[self.clb.scaling_group_spec()],
+                max_entities=scaling_group_max_entities,
+            )
+
+            self.scaling_group = ScalingGroup(
+                group_config=scaling_group_body,
+                pool=self.pool
+            )
+
+            self.first_scaling_policy = ScalingPolicy(
+                scale_by=24,
+                scaling_group=self.scaling_group
+            )
+
+            # If we didn't have this policy, then this test would always pass
+            # on an Otter deployment w/out Convergence enabled.  Reason: we
+            # start off with 24 servers.  We OOB-delete 2 of them, leaving 22
+            # actually running; however, Otter will still think 24 exist.  We
+            # scale up by one, and:
+            #
+            # - If Otter doesn't converge, then it'll just spin up another
+            # server, and will meet the test criteria, OR,
+            #
+            # - If Otter converges, it's entirely possible for it to notice
+            # that two servers are gone, for it to attempt to provision
+            # replacements, and then provision the third and final server, all
+            # before we get a response back from
+            # scaling_group.get_scaling_group_state.
+            #
+            # So, we need an intermediate step that *guarantees* our test the
+            # ability to inspect Otter's behavior.  This is that step.
+            self.second_scaling_policy = ScalingPolicy(
```

Comment:
Non-blocking - apologies, I don't think I understand this explanation very well - wouldn't scaling up by 1, waiting for those 2 IDs to be removed first, then waiting for there to be 25 servers be sufficient?  If otter does not have convergence, those 2 IDs will still be in the active list and never removed.

That's what `test_reaction_to_oob_deletion_then_scale_up` does, right?

Not that this breaks the test - was just wondering if it was necessary.  Again, we can file a cleanup ticket to remove the extra call, if this is a valid comment.